---
title: "Morgan's paper"
---

##Load data
  
Input metadata for Morgan's paper downloaded from https://huttenhower.sph.harvard.edu/sites/default/files/Pouchitis2015_Metadata_1.txt
OTU table and taxonomic table from https://huttenhower.sph.harvard.edu/sites/default/files/otu_table.txt
```{r}
studymetadata<- read.delim("~/Desktop/Morgan's/Pouchitis2015_Metadata_1.txt")
studyotutable<- read.delim("~/Desktop/Morgan's/otu_table.txt")
studytax<-studyotutable[,ncol(studyotutable)]
otutable<-studyotutable[,-c(1,ncol(studyotutable))]
library(tidyr)
studytax<-data.frame(studytax)
studytaxtable<-separate(studytax,studytax,into=c("k","p","c","o","f","g"),sep = ";")
```  
Keep only the 255 samples appeared in both metadata and OTU table, and create a phyloseq object with metadata, OTU table and tax table of those samples.
```{r}
library(phyloseq)
metadata16sid<-as.character(studymetadata$X16S_ID)#273,N00002
otutable_intersect<-otutable[,names(otutable) %in% metadata16sid]
phyotu<-as(otutable_intersect,"matrix")
rownames(phyotu)<-rownames(otutable_intersect)
META<-subset(studymetadata,X16S_ID %in% colnames(otutable_intersect))
rownames(META)<-META$X16S_ID
META<-sample_data(META)
phyotu<-phyotu[,rownames(META)]
OTU<-otu_table(phyotu, taxa_are_rows = TRUE)
phytax<-as(studytaxtable,"matrix")
rownames(phytax)<-rownames(OTU)
TAX<-tax_table(phytax)
sphy<-phyloseq(OTU,TAX,META)
```  

Remove OTUs without at least 3 counts in at least 3 samples.
```{r}
sphy1<-filter_taxa(sphy, function(x) sum(x>=3)>=3, TRUE)
# sphy2<-transform_sample_counts(sphy1,function(x) x/sum(x))
# sphyst1<-filter_taxa(sphy2, function(x) {sum(x>=0.05)>0},TRUE)
# sphy2t<-filter_taxa(sphy2, function(x) {mean(x)>=0.005},TRUE)
```

```{r}
# o2<-otu_table(sphy1)
# t2<-tax_table(sphy1)
# t2<-apply(t2,1:2,function(x){if (is.na(x)) x<-"unknown" else x<-x})
# df<-data.frame(o2,t2)
# tmpg<-numeric(2285)
# s<-sum(o2)
# for (i in 1:2285){
#   taxa<-df[i,"g"]
#   st<-sum(subset(df,g==taxa)[,1:255])
#   tmpg[i]<-st/s
# }
# sphyg<-tax_glom(sphy1,"p")
# sphygra<-transform_sample_counts(sphyg,function(x) x/sum(x))
# sphyraf<-filter_taxa(sphygra,function(x) {mean(x)>=0.005 && sum(x>=0.05)>0}, TRUE)
# taxaremain<-tax_table(sphyraf)[,"p"]
# sphyt = subset_taxa(sphy1, p %in% taxaremain)
```
  
Load transcriptome data (19908 genes) downloaded from https://bitbucket.org/biobakery/pouchitis/downloads/ and keep only the aforementioned 255 samples.
```{r}
gene_19908<-read.table("~/Desktop/Morgan's/biobakery-pouchitis-4e2bb7ca22ec/input/gene_expression.tab",header = TRUE)
rownames(gene_19908)<-gene_19908$Gene
#GSMxxxx labels for the 255 samples 
GSM2pouch<-read.csv("~/Desktop/Morgan's/GSM2pouch.csv")[,2:3]
GSM2pouch<-subset(GSM2pouch,X.1 %in% META$Patient_ID)
gene<-gene_19908[,GSM2pouch$X.2]
```  

```{r}

```
